Tests/BasicsTests/FileSystem/PathShimTests.swift

```diff
@@ -12,64 +12,65 @@
 
 import Basics
 import Foundation
-import XCTest
+import Testing
 
-class PathShimTests: XCTestCase {
-    func testRescursiveDirectoryCreation() {
-        // For the tests we'll need a temporary directory.
+struct PathShimTests {
+    @Test
+    func rescursiveDirectoryCreation() {
         try! withTemporaryDirectory(removeTreeOnDeinit: true) { path in
             // Create a directory under several ancestor directories.
             let dirPath = path.appending(components: "abc", "def", "ghi", "mno", "pqr")
             try! makeDirectories(dirPath)
 
             // Check that we were able to actually create the directory.
-            XCTAssertTrue(localFileSystem.isDirectory(dirPath))
+            #expect(localFileSystem.isDirectory(dirPath))
 
             // Check that there's no error if we try to create the directory again.
-            try! makeDirectories(dirPath)
+            #expect(throws: Never.self) {
```

Comment:
Just throw the error.

Reply:
You should probably never catch thrown errors in tests unless the test is itself testing that an error should be thrown. In general, allow the error to propagate out so that the testing library (XCTest or Swift Testing) can gather optimal metadata for it.